packages/blade/src/components/Button/Button/Button.tsx

```diff
@@ -36,22 +39,28 @@ type ButtonWithIconProps = ButtonCommonProps & {
 
 export type ButtonProps = ButtonWithoutIconProps | ButtonWithIconProps;
 
-const Button = ({
-  children,
-  icon,
-  iconPosition = 'left',
-  isDisabled = false,
-  isFullWidth = false,
-  isLoading = false,
-  onClick,
-  size = 'medium',
-  type = 'button',
-  variant = 'primary',
-  accessibilityLabel,
-}: ButtonProps): React.ReactElement => {
+const _Button: React.ForwardRefRenderFunction<BladeRefElement, ButtonProps> = (
+  {
+    children,
+    icon,
+    iconPosition = 'left',
+    isDisabled = false,
+    isFullWidth = false,
+    isLoading = false,
+    onClick,
+    size = 'medium',
+    type = 'button',
+    variant = 'primary',
+    accessibilityLabel,
+  },
+  ref,
+) => {
+  const buttonRef = useBladeInnerRef(ref);
```

Comment:
shouldn't this be part of `BaseButton` and not the Button component itself?

Reply:
Both are React components so we have to wrap React.ForwardRef on both Button & BaseButton. 

In Button (public facing) i've used useBladeInnerRef,
and for BaseButton i've directly exposed the native ref without any constraints.

Reply:
okay what i mean is in public facing API we have to use forwardref that is fine but `const buttonRef = useBladeInnerRef(ref);` this part can remain inside BaseButton only? and in this public facing component we can just take the ref and pass it to BaseButton just forward it as it is because BaseButton anyways is the place where the actual ref gets assigned to the native HTML button element

Reply:
I see, yeah that works. Same as https://github.com/razorpay/blade/pull/949#discussion_r1070920063 

Though just wanted to keep the BaseButton's ref more flexible and expose the ref as is since it's for internal use. 
Not a big deal tho, we can change later if requirements comes.